zerver/lib/message.py

```diff
@@ -260,11 +264,35 @@ def messages_for_ids(
     return message_list
 
 
+@overload
 def access_message(
     user_profile: UserProfile,
     message_id: int,
+    get_user_message: None = ...,
+    lock_message: bool = ...,
+) -> Message: ...
+@overload
+def access_message(
+    user_profile: UserProfile,
+    message_id: int,
+    get_user_message: Literal["exists"],
+    lock_message: bool = ...,
+) -> Tuple[Message, bool]: ...
+@overload
+def access_message(
+    user_profile: UserProfile,
+    message_id: int,
+    get_user_message: Literal["object"],
+    lock_message: bool = ...,
+) -> Tuple[Message, Optional[UserMessage]]: ...
```

Comment:
We really shouldn’t be casually introducing overloads like this: [mypy does not check the function body against overload signatures](https://zulip.readthedocs.io/en/latest/testing/mypy.html#using-overload-to-accurately-describe-variations), so they allow type errors to go unnoticed. These should be three separate functions instead.

Reply:
Hm -- TIL, OK.  I'll see if there's a clean way to split this into separate functions.

Reply:
#29712.